scripts/update_and_rebuild_mfem.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+#* This file is part of the MOOSE framework
+#* https://www.mooseframework.org
+#*
+#* All rights reserved, see COPYRIGHT for full restrictions
+#* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+#*
+#* Licensed under LGPL 2.1, please see LICENSE for details
+#* https://www.gnu.org/licenses/lgpl-2.1.html
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+for i in "$@"
+do
+  shift
+  if [[ "$i" == "-h" || "$i" == "--help" ]]; then
+    help=1;
+  fi
+
+  if [ "$i" == "--skip-submodule-update" ]; then
+    skip_sub_update=1;
+  else # Remove everything else before passing to cmake
+    set -- "$@" "$i"
+  fi
+done
+
+# Display help
+if [ -n "$help" ]; then
+  echo "Usage: $0 [-h | --help | --fast | --skip-submodule-update | <mfem cmake options> ]"
+  echo
+  echo "-h | --help              Display this message and list of available libmesh options"
+  echo "--skip-submodule-update  Do not update the libMesh submodule, use the current version"
+  echo
+  echo "Influential variables"
+  echo "CONDUIT_SRC_DIR          Path to conduit; default: ../framework/contrib/conduit from submodule"
+  echo "HDF5_DIR                 Path to hdf5; default PETSC_DIR"
+  echo "HYPRE_DIR                Path to hypre; default PETSC_DIR"
+  echo "LIBMESH_DIR              Path to libmesh (for netcdf); default ../libmesh/installed"
+  echo "METIS_DIR                Path to metis; default PETSC_DIR"
+  echo "MFEM_DIR                 MFEM install prefix; default: ../framework/contrib/mfem/installed"
+  echo "MFEM_SRC_DIR             Path to MFEM source; default: ../framework/contrib/mfem from submodule"
+  echo "PETSC_ARCH               PETSc arch; default: arch-moose if PETSC_DIR not set"
+  echo "PETSC_DIR                Path to PETSc install; default: ../petsc"
+  exit 0
+fi
+
+set -e
+
+if [ -n "$MFEM_SRC_DIR" ]; then
+  skip_sub_update=1
+else
+  MFEM_SRC_DIR="$(realpath -m "${SCRIPT_DIR}"/../framework/contrib/mfem)"
+fi
+MFEM_BUILD_DIR="${MFEM_SRC_DIR}/build"
+if [ -n "$MFEM_DIR" ]; then
+  echo "INFO: MFEM_DIR set - overriding default installed path"
+  echo "INFO: No cleaning will be done in specified path"
+else
+  MFEM_DIR="${MFEM_SRC_DIR}/installed"
+  rm -rf "$MFEM_DIR"
+fi
+
+CONDUIT_BUILD_DIR="${CONDUIT_SRC_DIR}/build"
+LIBMESH_DIR=${LIBMESH_DIR:-$(realpath -m "${SCRIPT_DIR}/../libmesh/installed")}
+if [ -z "$PETSC_DIR" ]; then
+  PETSC_DIR=$(realpath -m "${SCRIPT_DIR}/../petsc/arch-moose")
+fi
+HDF5_DIR=${HDF5_DIR:-$PETSC_DIR}
+HYPRE_DIR=${HYPRE_DIR:-$PETSC_DIR}
+METIS_DIR=${METIS_DIR:-$PETSC_DIR}
```

Comment:
For integration in MOOSE I don't think we should make these settable by the user. We don't want different versions of HDF5 or hypre or metis being used than what are used by PETSc